src/Appwrite/Auth/OAuth2/Disqus.php

```diff
@@ -0,0 +1,133 @@
+    {
+        $url = $this->endpoint . 'oauth/2.0/authorize/?' .
+            \http_build_query([
+                'response_type' => 'code',
+                'client_id' => $this->appID,
+                'state' => \json_encode($this->state),
+                'redirect_uri' => $this->callback,
+                'scope' => \implode(',', $this->getScopes())
+            ]);
+
+        return $url;
+    }
+
+    protected function getTokens(string $code): array
+    {
+        if (empty($this->tokens)) {
+            $this->tokens = \json_decode($this->request(
+                'POST',
+                $this->endpoint . 'oauth/2.0/access_token/',
+                ["Content-Type: application/x-www-form-urlencoded"],
+                \http_build_query([
+                    'grant_type' => 'authorization_code',
+                    "client_id" => $this->appID,
+                    "client_secret" => $this->appSecret,
+                    "redirect_uri" => $this->callback,
+                    'code' => $code,
+                    'scope' => \implode(' ', $this->getScopes()),
+                ])
+            ), true);
+        }
+        return $this->tokens;
+    }
+
+    public function refreshTokens(string $refreshToken): array
+    {
+        $this->tokens = \json_decode($this->request(
+            'POST',
+            $this->endpoint . 'oauth/2.0/access_token/?',
+            ['Content-Type: application/x-www-form-urlencoded'],
+            \http_build_query([
+                'grant_type' => 'refresh_token',
+                'refresh_token' => $refreshToken,
+                'client_id' => $this->appID,
+                'client_secret' => $this->appSecret,
+            ])
+        ), true);
+
+        if (empty($this->tokens['refresh_token'])) {
+            $this->tokens['refresh_token'] = $refreshToken;
+        }
+
+
+        return $this->tokens;
+    }
+
+    public function getUserID(string $accessToken): string
+    {
+        $user = $this->getUser($accessToken);
+
+        $userId = $user["response"]["id"];
+
+        return $userId;
+    }
+
+    public function getUserEmail(string $accessToken): string
+    {
+        $user = $this->getUser($accessToken);
+
+        $userEmail = $user["response"]["email"];
+
+        return $userEmail;
+    }
+
+    public function isEmailVerified(string $accessToken): bool
+    {
+        $user = $this->getUser($accessToken);
+
+        $isVerified = $user["response"]["isAnonymous"];
+
+        return $isVerified;
+    }
+
+    public function getUserName(string $accessToken): string
+    {
+        $user = $this->getUser($accessToken);
+
+        $username = $user["response"]["username"] ?? '';
+
+        return $username;
+    }
+
+    protected function getUser(string $accessToken): array
+    {
+        if (empty($this->user)) {
+            $user = $this->request(
+                'GET',
+                $this->endpoint . '3.0/users/details.json?' . \http_build_query([
```

Comment:
According to their docs,
 This endpoint is currently in[ beta version](https://disqus.com/api/docs/users/details/) . So this may need another PR. But that's something for the future.

Currently it returns this user object

```json
{
  "code": 0,
  "response": {
    "isFollowing": false,
    "disable3rdPartyTrackers": false,
    "isPowerContributor": false,
    "isFollowedBy": false,
    "isPrimary": true,
    "id": "385928894",
    "numFollowers": 0,
    "rep": 1.231755,
    "numFollowing": 0,
    "numPosts": 0,
    "location": "",
    "isPrivate": false,
    "joinedAt": "2022-07-08T18:09:18",
    "email": "jekneyagni@vusra.com",
    "username": "jekney",
    "isBlocking": false,
    "numLikesReceived": 0,
    "reputationLabel": "Average",
    "about": "",
    "name": "jekney",
    "url": "",
    "isBlocked": false,
    "numForumsFollowing": 0,
    "profileUrl": "https://disqus.com/by/jekney/",
    "reputation": 1.231755,
    "avatar": {
      "permalink": "https://disqus.com/api/users/avatars/jekney.jpg",
      "xlarge": {
        "permalink": "https://disqus.com/api/users/avatars/jekney.jpg",
        "cache": "//a.disquscdn.com/1656925438/images/noavatar128.png"
      },
      "cache": "//a.disquscdn.com/1656925438/images/noavatar92.png",
      "large": {
        "permalink": "https://disqus.com/api/users/avatars/jekney.jpg",
        "cache": "//a.disquscdn.com/1656925438/images/noavatar92.png"
      },
      "small": {
        "permalink": "https://disqus.com/api/users/avatars/jekney.jpg",
        "cache": "//a.disquscdn.com/1656925438/images/noavatar32.png"
      },
      "isCustom": false
    },
    "signedUrl": "",
    "isAnonymous": false
  }
}
```

Reply:
I notice `name` is `jekney` which looks like a nickname. Do docs state this is first and last name? If not, is there a different endpoint or parameter for that? 

Let me know if not, I will check if we should return empty string instead, or go with a nickname.

Reply:
This is the fake account created by me. So `jekney` is the full name here